packages/aws-cdk-lib/aws-iam/lib/role.ts

```diff
@@ -697,6 +697,20 @@ export class Role extends Resource implements IRole {
     return getPrecreatedRoleConfig(this);
   }
 
+  /**
+   * Adds given list of Principals to the TrustRelationship.
+   * This helps in overcoming how grantAssumeRole() works for Service / Account Principals.
+   * More Details on how grantAssumeRole() doesn't adds principals to trust relationship on below issue:
+   * https://github.com/aws/aws-cdk/issues/22550#issuecomment-1283095003
+   */
+  public addPrincipalsToAssumedBy(principals: IPrincipal[]): void {
```

Comment:
i think we can do it currently with the help of `role.assumeRolePolicy?` already, don't see any added value with this function.

Reply:
Correct, but I personally fell rather than relying **role.assumeRolePolicy**, it is better to have similar function like `grant()`, `grantAssume()`. Having function of similar kind **addPrincipalsToAssumedBy()** to give better helper value to users (though it is documented well).
Consider this following scenario, `grant()` is sufficient enough to add a PolicyDocument, but we do have helper functions like **grantSendMessage()** for `sqs`, **grantEncryptDepcrypt()** for `kms` to service for specific purposes right?, so this will also serve similar value is my thought.

Would like you hear from you..

Reply:
grant methods are encapsulation for policy statements, are we encapsulating any policies here(just the assume role), we are using the same object and just modifying it in the function, i'll  put this into perspective with the help of dx to make it more clear and to be sure i'm not missing something here:

Current method:
```
role.assumeRolePolicy?.addStatements(new iam.PolicyStatement({
  actions: ['sts:AssumeRole'],
  principals: [
    new iam.AccountPrincipal('123456789'),
    new iam.ServicePrincipal('beep-boop.amazonaws.com')
    ],
}));
```

Proposed: 

```
role.addPrincipalsToAssumedBy([
  new iam.AccountPrincipal('123456789012'),
  new iam.ServicePrincipal('beep-boop.amazonaws.com')
]);
```

Reply:
Exactly, you got my point. that **sts:AssumeRole** is the one i am talking about in terms of easing usage of utilities.
For Example see here how it is used in other utility functions in current scenario,

https://github.com/aws/aws-cdk/blob/975df1f5a17e9a2ff2b34223c84dd43e72057de2/packages/aws-cdk-lib/aws-kms/lib/key.ts#L201-L203

https://github.com/aws/aws-cdk/blob/975df1f5a17e9a2ff2b34223c84dd43e72057de2/packages/aws-cdk-lib/aws-kms/lib/private/perms.ts#L26-L28

This just adds `kms:Decrypt` to the provided `Grantee`.